Added Board Support for Starfive VisionFive2 RISC-V Single Board Computer

pfarwsi opened the pull request:
These list of patches contain Board support files for running Zephyr on the Starfive VisionFive2 RISC-V Single Board Computer based on JH7110 SOC. It's a basic support, no drivers other than UART have been tested.

kanakshilledar committed:
boards: riscv: visionfive2_jh7110: added board files for visionfive2

added basic board support files for visionfive2 SBC based on JH7110
RISC-V SoC.

Signed-off-by: Kanak Shilledar <kanakshilledar111@protonmail.com>

pfarwsi committed:
boards: riscv: visionfive2_jh7110: added board files for visionfive2

	added basic board support files for visionfive2 SBC based on JH7110
	RISC-V SoC. Added documentation for building and flashing.

Signed-off-by: Pratik Farkase <pratik.farkase@wsisweden.com>

pfarwsi committed:
dts: riscv: starfive: Added device trees for JH7110

Added basic device tree include files for Starfive JH7110.
The device tree contains basic uart support.

Signed-off-by: Pratik Farkase <pratik.farkase@wsisweden.com>

pfarwsi committed:
dts: bindings: serial: Added Synopsys Designware UART binding

Added missing Synopsys Designware UART YAML binding required by
JH7110 SOC.

Signed-off-by: Pratik Farkase <pratik.farkase@wsisweden.com>

pfarwsi committed:
soc: riscv: jh71xx: Added support for jh7110

Added support for JH7110 SOC. Added the kconfig support,
defconfig and linker files.

Signed-off-by: Pratik Farkase <pratik.farkase@wsisweden.com>

pfarwsi committed:
CODEOWNERS: Update maintainers/codeowners for Adding
support for VisionFive2

Added board support file support for Starfive VisionFive2
RISC-V Single Board Computer based on JH7110 SOC

Signed-off-by: Pratik Farkase <pratik.farkase@wsisweden.com>

kanakshilledar committed:
CODEOWNERS: update code owners

updated codeowners file to include
@kanakshilledar

Signed-off-by: Kanak Shilledar <kanakshilledar111@protonmail.com>

nordicjm reviewed (changes requested):
Zephyr is migrating to hwmv2 and will not be accepting hwmv1 boards or socs after this change, this submission should be converted to hwmv2

kanakshilledar commented:
@nordicjm Can you please provide documentation for the HWMv2? We have followed the [board porting guide](https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html) in zephyr docs for this PR. Can you please clarify on the HWMv2 and what needs to be changed?
Thanks

nordicjm commented:
See https://github.com/zephyrproject-rtos/zephyr/pull/68792 for documentation

kanakshilledar committed:
boards: riscv: visionfive2_jh7110: add board.yml for HWMv2

update patches required by newer HWMv2. implemented minimal
`board.yml` for VisionFive2

Signed-off-by: Kanak Shilledar <kanakshilledar111@protonmail.com>

kanakshilledar committed:
boards: riscv: visionfive2_jh7110: update according HWMv2

added `Kconfig.visionfive2_jh7110` required by HWMv2 styling of
board porting guide. Contains shared Kconfig variables.

Signed-off-by: Kanak Shilledar <kanakshilledar111@protonmail.com>

nordicjm reviewed (changes requested):
Changes need to be squashed into the main commits, retarget the branch to `collab-hwm`

kanakshilledar committed:
Merge branch 'zephyrproject-rtos:main' into main

fkokosinski commented on boards/riscv/visionfive2_jh7110/visionfive2_jh7110.dts:
```diff
@@ -0,0 +1,30 @@
+
+/ {
+        model = "StarFive VisionFive V2";
+        compatible = "starfive,visionfive-v2", "starfive,jh7110";
```
Please move the `"starfive,jh7110"` compat to the SoC node.

kanakshilledar replied:
I have moved `model` and `compatible` properties from `boards/riscv/visionfive2/visionfive2_jh7110.dts` to `dts/riscv/starfive/jh7110-visionfive-v2.dtsi`

With reference from `hifive_unmatched` board devicetree.

fkokosinski commented on soc/riscv/starfive_jh71xx/jh71xx/linker.ld:
```diff
@@ -0,0 +1,8 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <arch/riscv/common/linker.ld>
```
Instead of adding this file, you should use the following line in a CMake script:
```set(SOC_LINKER_SCRIPT ${ZEPHYR_BASE}/include/zephyr/arch/riscv/common/linker.ld CACHE INTERNAL "")```

pfarwsi replied:
Removed the redundant `linker.ld` as `CMakeLists.txt` script already contains the `set(SOC_LINKER_SCRIPT ${ZEPHYR_BASE}/include/zephyr/arch/riscv/common/linker.ld CACHE INTERNAL "")`.

fkokosinski commented on dts/riscv/starfive/jh7110-visionfive-v2.dtsi:
```diff
@@ -0,0 +1,219 @@
+        reg = <0x0 0x8000000 0x1 0x0>;
+    };
+
+    soc {
```
Let's add a compatible string for the SoC node

pfarwsi replied:
Added the `compatible` and `model` node to the `soc` node.

fkokosinski reviewed (changes requested):
Thanks for the PR!

PTAL at the issues the compliance check job reported

kanakshilledar commented on boards/riscv/visionfive2_jh7110/visionfive2_jh7110_defconfig:
```diff
@@ -0,0 +1,24 @@
+# Serial Drivers
+CONFIG_SERIAL=y
+CONFIG_UART_INTERRUPT_DRIVEN=y
+CONFIG_UART_NS16550=y
```
I am referencing the `hifive_unmatched` board as the base for working on this port and I see UART enabled in the `hifive_unmatched_defconfig`

pfarwsi committed:
dts: riscv: starfive: Added compatible and model
Added the `model` and `compatible` property in
dts/riscv/starfive/jh7110-visionfive-v2.dtsi
to `/` node and `soc` node respectively.

Signed-off-by: Pratik Farkase <pratik.farkase@wsisweden.com>

pfarwsi committed:
soc: riscv: starfive_jh71xx: jh71xx: remove linker.ld
Removed the redundant linker.ld script as CMakeLists.txt
already has the setting `SOC_LINKER_SCRIPT`.
Kconfig.soc: added `select 64BIT` option.

Signed-off-by: Pratik Farkase <pratik.farkase@wsisweden.com>

kanakshilledar committed:
boards: riscv: visionfive2_jh7110: implement requested changes

Kconfig.board: removed the `select 64bit` option as it is added
by the SoC.
visionfive2_jh7110: removed the `compatible` and `model` properties
as it was added in the `starfive/jh7110-visionfive2-v2.dtsi`

Signed-off-by: Kanak Shilledar <kanakshilledar111@protonmail.com>

kanakshilledar commented:
@nordicjm and @fkokosinski we have migrated the board to HWMv2 (#69472) according to #68792
